apps/wallet/src/features/Explorer/Highlights/IndicatorDot.tsx

```diff
@@ -0,0 +1,80 @@
+import type { FC } from 'react';
+import type { ViewStyle } from 'react-native';
+import { StyleSheet, TouchableOpacity } from 'react-native';
+import type { SharedValue } from 'react-native-reanimated';
+import Animated, {
+	interpolate,
+	useAnimatedStyle,
+	withTiming,
+} from 'react-native-reanimated';
+
+const AnimatedHoverable = Animated.createAnimatedComponent(TouchableOpacity);
+
+type IndicatorDotStyleProps = ViewStyle;
+
+interface IndicatorDotProps {
+	index: number;
+	setActiveIndex: (index: number) => void;
+	inputRange: number[];
+	outputRange: IndicatorDotStyleProps[];
+	animatedValue: SharedValue<number>;
+}
+
+const IndicatorDot: FC<IndicatorDotProps> = ({
+	index,
+	setActiveIndex,
+	inputRange,
+	outputRange,
+	animatedValue,
+}) => {
+	const heightOutputRange = outputRange.map((output) => output.height);
+	const opacityOutputRange = outputRange.map(() => 0.2);
+
+	heightOutputRange[index] = 40;
+	opacityOutputRange[index] = 1;
+
+	const animatedStyle = useAnimatedStyle(() => {
+		const height = interpolate(
+			animatedValue.value,
+			inputRange,
+			heightOutputRange as number[],
+		);
+
+		const opacity = interpolate(
+			animatedValue.value,
+			inputRange,
+			opacityOutputRange as number[],
+		);
```

Comment:
I think that these interpolate could be refactored to this:
```typescript
interpolate(
  animated.value,
  [index - 1, index, index + 1],
  [0.2, 1, 0.2],
)
```
and it doesn't need to know about the array of indicator